pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -943,6 +944,29 @@ func newAWSInstance(ec2 EC2, awsID, nodeName, availabilityZone, instanceType str
 	return self
 }
 
+// newAwsInstanceFromMetadata would build a new awsInstance from the metadata service,
```

Comment:
I left this one because this is what we want to do, but can't because of the issue with private dns names later on.  Eventually I hope we'll move NodeName to be the instance id, and then we can go back to this code.  But you're right, it's too big to leave.